### DISCUSSION TITLE
PWA offline after app restart

### DISCUSSION THREAD

**Discussion by emanuelef1** [ionic-team/capacitor#3205] [Q&A] [answered]
Hi everyone

not sure this is the right place but I wanted to ask something about **offline apps**.
I have an **Angular PWA with service worker** and we use Capacitor to bring the app on iOS and Android. 
The app works fine offline until you kill the app completely on the phone, if you put in background and reopen everything works fine. 
Then if you are in airplane mode tha app stays on the splash screen for a while and after a bit a white page is shown, but what has been downloaded earlier doens't work. 

I noticed in Android Capacitor calls the following mehtod to load the view: 
` webView.loadUrl(appUrl);`

and it's called every time the app is loaded, so I was wondering whether there is a way to kill the app and still have the app working offline.

Thanks everyone

**Comment by jcesarmobile** [comment, marked as answer]
The question is confusing because you say it's a PWA, but then mention `webView.loadUrl(appUrl);`.
So are you running it from Safari/Chrome or as native iOS/Android app.

If the later, Capacitor uses WKWebView on iOS, which doesn't support service workers. And on Android they work on the WebView, but doesn't work well when using Capacitor.
Capacitor apps don't need service workers for offline resources since the app ships with all the code bundled, so you should probably disable the service worker for the native build.
For Android, there is a workaround if you want the service worker to work, that will just pick the local files, see https://github.com/ionic-team/capacitor/issues/1655#issuecomment-579229390

**Reply by emanuelef1** [reply]
Hi, thank you for your reply and sorry for not be clear enough. 

We developed a PWA app in Angular and added Capacitor to bring on iOS and Android. 

The problem is related to the app run natively on iOS and Android. 

So the question would be:
does the app always need connection when launch the app after is closed from the task manager?
Is there any way to skip the connection at launch and show what has been stored in the cache?

To mention: after the app has started we see that some data has been stored, so the cache is working fine. And if we put the mobile in offline mode we can access data already downloaded, so the problem is only when the app starts again after being killed.

**Reply by jcesarmobile** [reply]
No, Capacitor apps don't need connection to work. The problem might be related to the service workers, as I said, they don't work on iOS and on Android they need the workaround to work, but the workaround will only allow to load the assets in public folder, I don't think cached assets will work.

**Reply by emanuelef1** [reply]
Thanks @jcesarmobile , we actually fixed the issue in our app removing the`server.url`setting from the `capacitor.config.json`. It turned out that we were using the live update, which requires a connection to work fine. 
Thanks again
